### DISCUSSION TITLE
5G UERANSIM setup

### DISCUSSION THREAD

**Discussion by HesselTjeerdsma** [magma/magma#14974] [Q&A] [answered]
Hi, is there anybody who could share their 5G SA UERANSIM config files? Our GNB is succesfully able to connect to the AGW, however the UE is not able to get an IP. In the mme.log, I see that the amf is unable to decode the PDU. Anybody experienced this before? Thank you!

**Comment by jordanvrtanoski** [comment]
I had used the 5G SA UERANSIM and had only an issue with a bug that used to be in 1.8, which is now fixed. If you are using docker deployment, the allocation of the IP will have issues if you had not restarted the containers (sometimes 2 or 3 restarts) after they had been created initially .

**Reply by jblakley** [reply]
@jordanvrtanoski, can you point me to the UERANSIM docs? I'm trying to get a real 5G SA UE to connect and running into issues. Pixel 7 seems to connect then drops after 10 seconds. I also have 2 Oneplus 9 Pros but they don't even see the network (think they may be locked to anything but AT&T, T-Mobile & Verizon). Trying with a simulated UE might be worthwhile just to make sure everything else is good. gNB and AGW seem to be working/connecting just fine.

**Comment by jordanvrtanoski** [comment, marked as answer]
You can get my version of the UERANSIM containing the configuration for Magma from [github](https://github.com/jordanvrtanoski/UERANSIM).

**Reply by User2707** [reply]
Hi @jordanvrtanoski.

I have deployed AGW using vagrant. gNB is successfully connected to the gateway. But while UE tries to connect amf gives the same error of 'Failed to decode PDU'.

Any suggestions?

**Reply by jordanvrtanoski** [reply]
Can you check with tshark if the messages arriving on eth1 are proper N1 messages?

**Reply by User2707** [reply]
Thanks for the reply @jordanvrtanoski 
Im attaching the amf logs and eth1 pcap file below
![image](https://github.com/magma/magma/assets/148062117/3957031f-cd01-447c-b682-77b5b557e344)
[eth1-capture.zip](https://github.com/magma/magma/files/13826454/eth1-capture.zip)

please let me know if u find anything.
Thank You.

**Reply by User2707** [reply]
Hi @jordanvrtanoski,
Thanks for the support.

I solved this issue, there was configuration mismatch between the UE and AMF configuration.
chnaged the sd and sst values corresponding to my AMF and PDU session successfully established.
